translator/src/main/java/com/google/devtools/j2objc/gen/StatementGenerator.java

```diff
@@ -873,7 +873,14 @@ public boolean visit(TypeLiteral node) {
     } else if (type.getKind().isPrimitive() || TypeUtil.isVoid(type)) {
       buffer.append(UnicodeUtils.format("[IOSClass %sClass]", TypeUtil.getName(type)));
     } else {
-      buffer.append(nameTable.getFullName(TypeUtil.asTypeElement(type))).append("_class_()");
+      TypeElement typeElement = TypeUtil.asTypeElement(type);
```

Comment:
I think i would put the comments // MIREGO ... outside all lines that were changed